src/components/Auth/Auth.tsx

```diff
@@ -276,14 +277,17 @@ function EmailAuth({
         if (signInError) setError(signInError.message)
         break
       case 'sign_up':
-        const { error: signUpError } = await supabaseClient.auth.signUp(
+        const { error: signUpError, data: signUpData } = await supabaseClient.auth.signUp(
           {
             email,
             password,
           },
           { redirectTo }
         )
         if (signUpError) setError(signUpError.message)
+        // checking if it has access_token to know if email verification is disabled 
+        else if (!signUpData.hasOwnProperty('access_token')) 
```

Comment:
It might be better to check for `confirmation_sent_at` as they will more precisely tell you if email verification is turned on.

Reply:
Sure, will update it now.